src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs

```diff
@@ -1407,15 +1402,7 @@ public virtual void OnActionExecuted(ActionExecutedContext context)
                 throw new ArgumentNullException(nameof(prefix));
             }
 
-            if (BindingContext == null)
-            {
-                var message = Resources.FormatPropertyOfTypeCannotBeNull(
-                    nameof(BindingContext),
-                    typeof(Controller).FullName);
-                throw new InvalidOperationException(message);
-            }
-
-            return TryUpdateModelAsync(model, prefix, BindingContext.ValueProvider);
+            return TryUpdateModelAsync(model, prefix, new CompositeValueProvider(ControllerContext.ValueProviders));
```

Comment:
That's the nice part of just having one context. The unit testing is really simple now.